### DISCUSSION TITLE
[16일차 - 지원] Arrow function을 언제 쓰고, 기존의 function과 무엇이 다른가요?

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#51] [프론트 면접 질문]
arrow function을 언제 쓰고, 기존의 function과 무엇이 다른가요?

**Comment by ez1n** [comment]
arrow function (화살표함수)은 ES6에 도입된 문법으로 화살표를 이용하여 간편하게 함수를 선언할 수 있는 방식입니다.

기존 function에 비해 function 키워드를 생략하거나 매개변수가 하나인 경우 괄호를 생략하는 등 함수를 간결하게 표현할 수 있고 바디가 표현식 하나로 이루어진 경우 화살표 뒤의 값을 그대로 리턴할 수 있다는 특징이 있습니다.
특히 상위 스코프의 this 값을 그대로 가져올 수 있다는 특징이 있는데 화살표 함수 내에서의 this는 자신이 선언된 위치의 this(lexical this)를 가리키며 call, apply, bind 메서드를 사용하여 this를 변경할 수 없습니다. 때문에 메서드, 생성자, 이벤트 리스너의 콜백 함수로는 사용할 수 없지만 일반적인 콜백함수에서 유용하게 사용될 수 있습니다.
